src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -615,30 +467,30 @@ public SolutionState RemoveProject(ProjectId projectId)
         /// Creates a new solution instance with the project specified updated to have the new
         /// assembly name.
         /// </summary>
-        public SolutionState WithProjectAssemblyName(ProjectId projectId, string assemblyName)
+        public (SolutionState, ProjectState) WithProjectAssemblyName(ProjectId projectId, string assemblyName)
         {
             var oldProject = GetRequiredProjectState(projectId);
             var newProject = oldProject.WithAssemblyName(assemblyName);
 
             if (oldProject == newProject)
             {
-                return this;
+                return (this, oldProject);
             }
 
-            return ForkProject(newProject, new CompilationAndGeneratorDriverTranslationAction.ProjectAssemblyNameAction(assemblyName));
+            return ForkProject(newProject);
```

Comment:
the translation actions are moved to the callsite in SolutionCompilationTracker.  I origially moved all these methods one at a time, so i have high confidence i didn't miss things.